DisCatSharp/Clients/Rest/GuildMethods/ApplicationCommands/DiscordApiClient.GuildMethods.ApplicationCommands.cs

```diff
@@ -0,0 +1,255 @@
+			DmPermission = command.DmPermission/*,
+			Nsfw = command.IsNsfw*/
+		};
+
+		var route = $"{Endpoints.APPLICATIONS}/:application_id{Endpoints.GUILDS}/:guild_id{Endpoints.COMMANDS}";
+		var bucket = this.Rest.GetBucket(RestRequestMethod.POST, route, new {application_id = applicationId, guild_id = guildId }, out var path);
+
+		var url = Utilities.GetApiUriFor(path, this.Discord.Configuration);
+		var res = await this.ExecuteRestRequest(this.Discord, bucket, url, RestRequestMethod.POST, route, payload: DiscordJson.SerializeObject(pld));
+
+		var ret = JsonConvert.DeserializeObject<DiscordApplicationCommand>(res.Response);
+		ret.Discord = this.Discord;
+
+		return ret;
+	}
+
+	/// <summary>
+	/// Gets a guild application command.
+	/// </summary>
+	/// <param name="applicationId">The application id.</param>
+	/// <param name="guildId">The guild id.</param>
+	/// <param name="commandId">The command id.</param>
+	internal async Task<DiscordApplicationCommand> GetGuildApplicationCommandAsync(ulong applicationId, ulong guildId, ulong commandId)
+	{
+		var route = $"{Endpoints.APPLICATIONS}/:application_id{Endpoints.GUILDS}/:guild_id{Endpoints.COMMANDS}/:command_id";
+		var bucket = this.Rest.GetBucket(RestRequestMethod.GET, route, new {application_id = applicationId, guild_id = guildId, command_id = commandId }, out var path);
+
+		var url = Utilities.GetApiUriFor(path, this.Discord.Configuration);
+		var res = await this.ExecuteRestRequest(this.Discord, bucket, url, RestRequestMethod.GET, route);
+
+		var ret = JsonConvert.DeserializeObject<DiscordApplicationCommand>(res.Response);
+		ret.Discord = this.Discord;
+
+		return ret;
+	}
+
+	/// <summary>
+	/// Edits a guild application command.
+	/// </summary>
+	/// <param name="applicationId">The application id.</param>
+	/// <param name="guildId">The guild id.</param>
+	/// <param name="commandId">The command id.</param>
+	/// <param name="name">The name.</param>
+	/// <param name="description">The description.</param>
+	/// <param name="options">The options.</param>
+	/// <param name="nameLocalization">The localizations of the name.</param>
+	/// <param name="descriptionLocalization">The localizations of the description.</param>
+	/// <param name="defaultMemberPermission">The default member permissions.</param>
+	/// <param name="dmPermission">The dm permission.</param>
+	/// <param name="isNsfw">Whether this command is marked as NSFW.</param>
+	internal async Task<DiscordApplicationCommand> EditGuildApplicationCommandAsync(ulong applicationId, ulong guildId, ulong commandId,
+		Optional<string> name, Optional<string> description, Optional<IReadOnlyCollection<DiscordApplicationCommandOption>> options,
+		Optional<DiscordApplicationCommandLocalization> nameLocalization, Optional<DiscordApplicationCommandLocalization> descriptionLocalization,
+		Optional<Permissions> defaultMemberPermission, Optional<bool> dmPermission, Optional<bool> isNsfw)
+	{
+		var pld = new RestApplicationCommandEditPayload
```

Comment:
Is this duplicated?????

Reply:
Ah I see the `slight` deviation where this includes the `applicationId`. I knew I wasn't going crazy lol.